synapseclient/core/utils.py

```diff
@@ -342,12 +344,38 @@ def is_same_base_url(url1: str, url2: str) -> bool:
 def is_synapse_id_str(obj):
     """If the input is a Synapse ID return it, otherwise return None"""
     if isinstance(obj, str):
-        m = re.match(r"(syn\d+$)", obj)
+        m = re.match(r"(syn\d+(\.\d+)?$)", obj)
         if m:
             return m.group(1)
     return None
 
 
+def get_synid_and_version(
```

Comment:
Nit: This would have been a good opportunity to practice test driven development: https://testdriven.io/blog/modern-tdd/ .  It's not the easiest to switch to, but it's more important to ensure that you've thought about the input / output of your function and you contribute unit tests along with your functions.